src/main/java/com/swervedrivespecialties/exampleswerve/commands/DriveCommand.java

```diff
@@ -2,22 +2,28 @@
 
 import com.swervedrivespecialties.exampleswerve.Robot;
 import com.swervedrivespecialties.exampleswerve.subsystems.DrivetrainSubsystem;
-import edu.wpi.first.wpilibj.command.Command;
+import com.swervedrivespecialties.exampleswerve.util.util;
+
 import edu.wpi.first.wpilibj.geometry.Translation2d;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+
 import org.frcteam2910.common.robot.Utilities;
 
-public class DriveCommand extends Command {
+public class DriveCommand extends CommandBase {
+
+    private static DrivetrainSubsystem _drive;
 
-    public DriveCommand() {
-        requires(DrivetrainSubsystem.getInstance());
+    public DriveCommand(DrivetrainSubsystem drive) {
+        _drive = drive;
+        addRequirements(_drive);
     }
 
     @Override
-    protected void execute() {
+    public void execute() {
         double forward = -Robot.getOi().getPrimaryJoystick().getRawAxis(1);
         forward = Utilities.deadband(forward);
         // Square the forward stick
-        forward = Math.copySign(Math.pow(forward, 2.0), forward);
+        forward = util.getSpeedScale(_drive.getMinSpeedScale(), 0) * Math.copySign(Math.pow(forward, 2.0), forward);
```

Comment:
You use this construction 3 times, and without explanation it's nonsense. util.getSpeedScale(_drive.getMinSpeedScale(), 0) just returns _drive.getMinSpeedScale().  You do actually want to do this, but you want the additional speed scale to be the result from the right trigger. And, if you're doing it 3 times, make a double speedScale you define up top and use in all 3 places.

Reply:
I put it in util in case we used it in any other command but i see what you're saying, i changed it